datasources/src/main/java/org/opensearch/sql/datasources/transport/TransportPatchDataSourceAction.java

```diff
@@ -0,0 +1,74 @@
+/*
+ *
+ *  * Copyright OpenSearch Contributors
+ *  * SPDX-License-Identifier: Apache-2.0
+ *
+ */
+
+package org.opensearch.sql.datasources.transport;
+
+import static org.opensearch.sql.datasources.utils.XContentParserUtils.NAME_FIELD;
+import static org.opensearch.sql.protocol.response.format.JsonResponseFormatter.Style.PRETTY;
+
+import org.opensearch.action.ActionType;
+import org.opensearch.action.support.ActionFilters;
+import org.opensearch.action.support.HandledTransportAction;
+import org.opensearch.common.inject.Inject;
+import org.opensearch.core.action.ActionListener;
+import org.opensearch.sql.datasource.DataSourceService;
+import org.opensearch.sql.datasources.model.transport.PatchDataSourceActionRequest;
+import org.opensearch.sql.datasources.model.transport.PatchDataSourceActionResponse;
+import org.opensearch.sql.datasources.service.DataSourceServiceImpl;
+import org.opensearch.sql.protocol.response.format.JsonResponseFormatter;
+import org.opensearch.tasks.Task;
+import org.opensearch.transport.TransportService;
+
+public class TransportPatchDataSourceAction
+    extends HandledTransportAction<PatchDataSourceActionRequest, PatchDataSourceActionResponse> {
+
+  public static final String NAME = "cluster:admin/opensearch/ql/datasources/patch";
```

Comment:
can you make it same as update? I see no point in having two update permissions.